### PR TITLE
Jv spring boot book store 4

### PR DESCRIPTION
**Pull request by Vlad1507**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,10 @@
 			<artifactId>liquibase-maven-plugin</artifactId>
 			<version>${liquibase.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>org.hibernate.validator</groupId>
+			<artifactId>hibernate-validator</artifactId>
+		</dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml:85]: better use spring starter validation

File: src/main/java/com/store/bookstore/mapper/BookMapper.java [hunk 2]
```diff
@@ -5,10 +5,15 @@
...
+import org.mapstruct.MappingTarget;
+import org.springframework.data.jpa.repository.Modifying;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
     BookDto toDto(Book book);
 
     Book toModel(CreateBookRequestDto requestDto);
+
+    @Modifying
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/mapper/BookMapper.java:17]: why do you need this annotation?

File: src/main/java/com/store/bookstore/validation/IsbnValidator.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class IsbnValidator implements ConstraintValidator<Isbn, String> {
+    private static final String ISBN_PATTERN = "(\\d{3})(\\d)(\\d{5})(\\d{3})(\\d)";
+
+    @Override
+    public boolean isValid(String isbn, ConstraintValidatorContext constraintValidatorContext) {
+        Matcher matcher = Pattern.compile(ISBN_PATTERN).matcher(isbn);
+        String formattedIsbn = matcher.replaceFirst("$1-$2-$3-$4-$5");
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/validation/IsbnValidator.java:14]: Why do you need it? Can it provide some exceptions?

File: src/main/java/com/store/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/store/bookstore/exception/CustomGlobalExceptionHandler.java:19]: Let’s  handle EntityNotFoundException and return 404 status code

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s improve your solution ;)

File: pom.xml [hunk 5]
```diff
@@ -79,6 +79,15 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
+			<version>3.4.0</version>
```
Comment (Elena-Bruyako) [inline pom.xml:89]:
```suggestion
```

**Review by Elena-Bruyako — APPROVED** [review]
One comment about version, you can fix it in the next task

**Pull request merged**